src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -322,13 +323,25 @@ internal void PublishRpcChannelReadyEvent(RpcEvent initEvent)
             }
         }
 
-        public void RegisterFunctions(IObservable<FunctionMetadata> functionRegistrations)
+        public void SetupFunctionInvocationBuffers(IEnumerable<FunctionMetadata> functions)
         {
-            if (functionRegistrations == null)
+            _functions = functions;
```

Comment:
@brettsam  - LanguageWorkerChannel now has a list of functions which can be used to look up function name from functionId on load response.
Also, this list gets reset on JobHost restart